2017 Robot/src/org/usfirst/frc/team4999/lights/animations/Blink.java

```diff
@@ -27,12 +28,12 @@ public Blink(Color[] colors, int[] waittimes) {
 	}
 
 	@Override
-	public Color[] animate(Color[] pixels) {
-		Color[] out = pixels.clone();
-		for(int i = 0; i < pixels.length; i++) {
-			out[i] = colors[idx];
+	public Packet[] animate() {
+		Packet out[] = {new Packet(0, colors[idx], 1, 1)};
+		idx++;
```

Comment:
As always, this pattern is better written as:
idx = (idx + 1) % colors.length;

Here and in all other places.